Swiftification/String.swift

```diff
@@ -53,6 +53,15 @@ public extension String {
         return result
     }
 
+    /// Returns a value with string inserted at index.
+    @warn_unused_result
+    func insert(string: String?, atIndex index: Int) -> String {
```

Comment:
I'd imagine an insert function would allow me to insert a string. I can't think of why insert would need to take a nil. If the caller has a nil it wants to conditionally insert, it seems like that's the caller's problem.